src/regex/match_data.cr

```diff
@@ -158,11 +158,8 @@ class Regex
       match = self[group_name]?
       unless match
         ret = LibPCRE.get_stringnumber(@code, group_name)
-        if ret < 0
-          raise KeyError.new("Capture group named '#{group_name}' does not exist")
-        else
-          raise KeyError.new("Capture group named '#{group_name}' is not matched")
-        end
+        status = ret < 0 ? "does not exist" : "was not matched"
+        raise KeyError.new("Capture group '#{group_name}' #{status}")
```

Comment:
I think having 2 if branches here is much cleaner. Removing duplication isn't always good.